src/openms/include/OpenMS/FORMAT/TriqlerFile.h

```diff
@@ -0,0 +1,222 @@
+        for (const OpenMS::Peak2D::IntensityType &intensity : intensities)
+        {
+          result += intensity;
+        }
+        return result;
+      }
+
+      OpenMS::Peak2D::IntensityType meanIntensity_(const std::set< OpenMS::Peak2D::IntensityType > &intensities) const
+      {
+        return sumIntensity_(intensities) / intensities.size();
+      }
+
+      class TriqlerLine_
+      {
+      public :
+        TriqlerLine_(
+            const String& run,
+            const String& condition,
+            const String& precursor_charge,
+            const String& search_score,           
+            const String& intensity,
+            const String& sequence,
+            const String& accession
+        ):  run_(run),
+            condition_(condition),
+            precursor_charge_(precursor_charge),
+            search_score_(search_score),
+            intensity_(intensity),
+            sequence_(sequence),
+            accession_(accession)
+             {}
+        
+        TriqlerLine_(TriqlerLine_&& m) = default;
+
+        TriqlerLine_(const TriqlerLine_& m) = default;
+
+        const String& run() const {return this->run_;}
+        const String& condition() const {return this->condition_;}
+        const String& charge() const {return this->precursor_charge_;}
+        const String& searchScore() const {return this->search_score_;}
+        const String& intensity() const {return this->intensity_;}
+        const String& sequence() const {return this->sequence_;}
+        const String& accession() const {return this->accession_;}
+
+        String toString() const
+        {
+          const String delim("\t");
+          return  run_
+                  + delim + condition_
+                  + delim + precursor_charge_
+                  + delim + search_score_
+                  + delim + intensity_
+                  + delim + sequence_
+                  + delim + accession_;
+        }
+
+        friend bool operator<(const TriqlerLine_ &l,
+                              const TriqlerLine_ &r) {
+
+          return std::tie(l.accession_, l.run_, l.condition_, l.precursor_charge_, l.intensity_, l.sequence_) <
+                 std::tie(r.accession_, r.run_, r.condition_, r.precursor_charge_, r.intensity_, r.sequence_);
+        }
+
+
+      private:
+        String run_;
+        String condition_;
+        String precursor_charge_;
+        String search_score_;
+        String intensity_;
+        String sequence_;
+        String accession_;
+      };
+
+      /*
+        *  @brief Constructs the lines and adds them to the TextFile
+        *  @param peptideseq_quantifyable Has to be a set (only) for deterministic  ordered output
+        */
+      template <class LineType>
+      void constructFile_(TextFile& csv_out,
+                          const std::set<String>& peptideseq_quantifyable,
+                          LineType & peptideseq_to_prefix_to_intensities) const;
```

Comment:
some formatting required here... there are some other occurrences of this

Reply:
this is a template, but no definition. This is trouble.
But: it seems only one type is actually needed, so why not remove the template?